src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/BlockStorageCommand.java

```diff
@@ -0,0 +1,75 @@
+package io.github.thebusybiscuit.slimefun4.core.commands.subcommands;
+
+import io.github.thebusybiscuit.slimefun4.core.commands.SlimefunCommand;
+import io.github.thebusybiscuit.slimefun4.core.commands.SubCommand;
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+import me.mrCookieSlime.CSCoreLibPlugin.Configuration.Config;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.ConsoleCommandSender;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Checks the contents of BlockStorage and creates a file of the contents.
+ *
+ * @author Sefiraat
+ */
+class BlockStorageCommand extends SubCommand {
+
+    @ParametersAreNonnullByDefault
+    BlockStorageCommand(Slimefun plugin, SlimefunCommand cmd) {
+        super(plugin, cmd, "blockstorage", false);
+    }
+
+    @Override
+    protected String getDescription() {
+        return "commands.blockstorage.description";
+    }
+
+    @Override
+    public void onExecute(CommandSender sender, String[] args) {
+        if (sender.hasPermission("slimefun.command.blockstorage") || sender instanceof ConsoleCommandSender) {
+            final Map<String, Integer> map = new LinkedHashMap<>();
+            for (World world : Bukkit.getWorlds()) {
+                for (Config value : BlockStorage.getRawStorage(world).values()) {
```

Comment:
this is pretty expensive and we're also writing this on the main thread. Can we move this to another?

Reply:
Not sure if it helps, but I had this run on CYT at it's peak of blockstorage with like 30mil blocks and it still took ms to run, and it's one that's only due to run very situationally? Cant see why I couldn't run it async however

Reply:
Both made async and moved into a dump subcommand. It means the completer only has one item for now, but it's at least future ready then.